### PR TITLE
[FCOS] non-virtual-ip: replace the script with podman wrapper

### PR DESCRIPTION
**Pull request by vrutkovs**
Replace Python script with a wrapper which runs the script in a podman container (using host networking).

Requires https://github.com/openshift/baremetal-runtimecfg/pull/46
Fixes https://github.com/openshift/machine-config-operator/issues/1430

Master counterpart: https://github.com/openshift/machine-config-operator/pull/1521

### CODE REVIEW

File: templates/worker/00-worker/baremetal/files/baremetal-non-virtual-ip.yaml [hunk 1]
```diff
@@ -3,136 +3,9 @@ mode: 0755
...
-        except SubnetNotFoundException:
-            sys.exit(1)
-
-    if __name__ == '__main__':
-        main()
+    #!/usr/bin/bash
+    podman run --rm \
+      --net=host \
+      --entrypoint=/usr/bin/non_virtual_ip \
+      -ti {{ .Images.BaremetalRuntimeCfgImage }} \
```
Comment (mandre) [inline templates/worker/00-worker/baremetal/files/baremetal-non-virtual-ip.yaml]: Have you tried without attaching a TTY? I believe we shouldn't need the `-ti` option.

**Comment by mandre** [comment]
This is going to conflict with https://github.com/openshift/machine-config-operator/pull/1483, can we wait until it merges?
I'd like to keep the port clean in #1483 and resolve the conflict in your PR if you don't mind.

**Comment by vrutkovs** [comment]
>This is going to conflict with #1483, can we wait until it merges?

`fcos` branch is independent of master, so we'll cherry-pick it back to master if the idea works

**Comment by mandre** [comment]
> > This is going to conflict with #1483, can we wait until it merges?
> 
> `fcos` branch is independent of master, so we'll cherry-pick it back to master if the idea works

Ah right, I forgot this PR was against a different branch. All good then.

**Comment by mandre** [comment]
This should be rebased after https://github.com/openshift/machine-config-operator/pull/1483.

Also, after re-syncing with master, you should leave out ovirt platform because they're still using the older version of `non-virtual-ip` script.

Note that there is now another python script `nodeip-finder` that should be moved to `baremetal-runtimecfg` the same way you did in https://github.com/openshift/baremetal-runtimecfg/pull/46.

**Comment by vrutkovs** [comment]
>Also, after re-syncing with master, you should leave out ovirt platform because they're still using the older version of non-virtual-ip script.

Are these incompatible? I was hoping one script could be reused for all platoforms.
cc @rgolangh

**Comment by vrutkovs** [comment]
Updated https://github.com/openshift/baremetal-runtimecfg/pull/46.

I didn't change the wrapper in `fcos` branch, as its still lagging behind master (it would be rebased soon)

**Comment by vrutkovs** [comment]
This now requires https://github.com/openshift/baremetal-runtimecfg/pull/48 (since fcos branch promotes to 4.4 for now)

**Comment by vrutkovs** [comment]
/cc @LorbusChris 

This should fix oVirt / Openstack not joining worker nodes, PTAL

**Comment by celebdor** [comment]
It looks good. I wonder how much delay it is introduced by podman pulling the images, but with a local registry it shouldn't be much.

**Comment by LorbusChris** [comment]
is this a real concern tho? After all we deliver all of OpenShift's operators as container images as well

**Comment by vrutkovs** [comment]
This script is used by NetworkManager, so potentially we may end up with misconfigured networking

**Comment by cgwalters** [comment]
Thanks a ton for doing this!  This also solves the problem of having substantial baremetal code inside the MCO, which is a double win.

**Comment by LorbusChris** [comment]
We should get this into master as well asap. Do we have OpenStack e2e tests here?
/approve

**Comment by vrutkovs** [comment]
/hold

Wrong script names

**Comment by vrutkovs** [comment]
/hold cancel

Update entrypoint, ready for review

**Comment by LorbusChris** [comment]
xref master PR: https://github.com/openshift/machine-config-operator/pull/1521
Let's wait for e2e testing on master to complete before merging here.
/approve

**Pull request merged**